pubsub/spec/pubsub_spec.rb

```diff
@@ -0,0 +1,71 @@
+# Copyright 2015 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ENV["RACK_ENV"] = "test"
+
+require_relative "../pubsub"
+require "spec_helper"
+
+describe "Pub/Sub sample" do
+  def app
+    Sinatra::Application
+  end
+
+  it "says hello" do
+    get "/"
+    expect(last_response).to be_ok
+    expect(last_response.body).to eq("Hello World!")
+  end
+
+  it "get topics" do
+    get "/topics"
+    expect(last_response).to be_ok
+    expect(last_response.body).to \
+      include(Sinatra::Application.settings.topic_name)
+  end
+
+  it "get subs" do
+    get "/subs"
+    expect(last_response).to be_ok
+    expect(last_response.body).to \
+      include(Sinatra::Application.settings.sub_name)
+    expect(last_response.body).to \
+      include(Sinatra::Application.settings.sub2_name)
+  end
+
+  it "takes a publish" do
+    post "/pub", "Message"
+    expect(last_response).to be_ok
+  end
+
+  it "can pull from a sub" do
+    post "/pub", "PullThis"
+    sleep 1
```

Comment:
This will result in flakey tests.

Instead of sleeping, attempt to pull a few times.  Eg. pull ever 100ms until the condition is met OR we time out (eg. after 5s)

I don't think RSpec has this baked in (people frequently have helper methods for this use-case).  So maybe something simple (but less flakey) like this (?)

```ruby
post "/publish", "my message"

5.times do
  get "/pull_messages"
  sleep 1000 unless last_response.body.include? "my message"
end

expect(last_response).to be_ok
expect(last_response.body).to include("my message")
```